Hangman attempted chars

RudraNilBasu opened the pull request:
Displays a list of characters attempted by the user.
* checked with resizing the screen and enabling the image.
* qsTr() is used for strings that are to be translated
* All types of characters will be included in the list, and not just A-Z English characters.
(Since whatever string comes to the processKeyPress(text) function as parameter is 
added to the list).

petitlapin commented on src/activities/hangman/hangman.js:
```diff
@@ -76,6 +76,7 @@ function initLevel() {
     var currentLesson = lessons[currentLevel];
     wordList = Lang.getLessonWords(dataset, currentLesson);
     Core.shuffle(wordList);
+    items.guessedText.text=qsTr("Attempted: ");
```
why is it initialized twice?

petitlapin commented on src/activities/hangman/hangman.js:
```diff
@@ -136,6 +137,9 @@ function processKeyPress(text) {
     alreadyTypedLetters.push(text);
 
+    // add
+    items.guessedText.text=items.guessedText.text.concat(text, ", ");
```
it's quite tricky but I would keep all the letters and rebuild the typed letters string using:
items.guessedText.text = qsTr("Attempted: %1").arg(usedLetters)
usedLetters can probably be built using alreadyTypedLetters and join (something like qsTr("Attempted: %1").arg(alreadyTypedLetters.join(", ")).
For translation purpose, it's better to have the whole context (we could have a language that would need to display it on the other direction: "used letters: Attempted" and if we do like now, we won't be able to do it as the code supposes that the letters should be displayed on right of "Attempted: ")

RudraNilBasu replied:
In the recent commit, I solved it in a similar way in the _form()_ method

In that method, I initialized a string with "Attempted: ". Then I sorted the "alreadyTypedLetters" 
array and inserted it into the forming string one by one. When the full traversal is over, I am 
returning the string after passing it through qsTr(), and that is shown in the top of the activity.

RudraNilBasu commented:
@petitlapin thank you for the review. I have updated the PR. These problems should be fixed 
* the warning related to the anchors is sorted out
* the layout is fixed, the text doesn't appear on top of the hints image. The text is placed on a Rectangle,
to enhance the visibility
* Characters in the text are sorted
* Fixed translation issues in a language that displays in the other direction: "used letters: Attempted"

petitlapin commented on src/activities/hangman/hangman.js:
```diff
@@ -136,6 +136,9 @@ function processKeyPress(text) {
     alreadyTypedLetters.push(text);
 
+    // add
+    items.guessedText.text=form()
```
form is not a good name for a function, maybe you can call it more explicitely.
Also, can you add a space before and after the '=' sign?

petitlapin commented on src/activities/hangman/hangman.js:
```diff
@@ -200,6 +203,8 @@ function initSubLevel() {
     alreadyTypedLetters = new Array();
     currentWord = items.goodWord.translatedTxt;
     items.hidden.text = ""
+    items.guessedText.text=qsTr("Attempted: ");
```
I would call form() here, and in form() instead of returning the string, I would directly update items.guessedText.text like stated below

RudraNilBasu commented:
@petitlapin Thank you for the review, I am away from keyboard for few days, will make the changes when I get back with it.

RudraNilBasu commented:
@petitlapin updated the PR, the issues are solved in this commit.

petitlapin commented:
merged in kde master branch, thank you!
